Simulation: enable returning of organized point clouds

damienjadeduff opened the pull request:
This pull request extends the functionality of the simulation module to return simulated point clouds in organised format as well as the default unorganised format.
The simulation module may not be complete but it is still usable for some tasks. We are using it for testing segmentation algorithms. 
Existing code would not need to be changed as the flag calling for organised point clouds defaults to the previous functionality.
Kind regards.

stefanbuettner commented on simulation/src/range_likelihood.cpp:
```diff
@@ -669,7 +670,7 @@ pcl::simulation::RangeLikelihood::getPointCloud (pcl::PointCloud<pcl::PointXYZRG
   float zn = z_near_;
   float zf = z_far_;
-
+  
```
Remove white spaces

stefanbuettner commented on simulation/src/range_likelihood.cpp:
```diff
@@ -702,12 +707,25 @@ pcl::simulation::RangeLikelihood::getPointCloud (pcl::PointCloud<pcl::PointXYZRG
       }
+      else if (organized)
+      {
+	pc->points[idx].z = std::numeric_limits<float>::quiet_NaN ();
```
Don't use tabs and watch indentation.

stefanbuettner commented:
Hey damien,

thanks for the PR. Looks good to me. Another step towards fully supporting organized point clouds :grinning:

Cheers,
Stefan

taketwo commented on simulation/src/range_likelihood.cpp:
```diff
@@ -702,12 +707,25 @@ pcl::simulation::RangeLikelihood::getPointCloud (pcl::PointCloud<pcl::PointXYZRG
+        pc->points[idx].y = std::numeric_limits<float>::quiet_NaN ();
+        pc->points[idx].b = 0;
+        pc->points[idx].g = 0;
+        pc->points[idx].r = 0;
```
Or just: `pc->points[idx].rgba = 0;`

damienjadeduff commented:
Hi there,
I'm glad it if can be of use to someone. I have fixed the white-space issues as per Stefan's observations, shortened the code as per Sergey's suggestion, and squashed it. 
Regards,
Damien

SergioRAgostinho commented:
Hey @damienjadeduff. Can I ask you do one more thing? The pointcloud is being set with the `is_dense` flag to false over [here](https://github.com/PointCloudLibrary/pcl/pull/1649/files#diff-26fdb26d67bcf5047b9a93d21060b603L663) by default.  Can you change it to `true` and only set it to `false` only if the code reaches [this section](https://github.com/PointCloudLibrary/pcl/pull/1649/files#diff-26fdb26d67bcf5047b9a93d21060b603R715).

damienjadeduff commented:
@SergioRAgostinho thanks. Fix pushed.

SergioRAgostinho commented:
Nop! Thank you! :) Keep contributing :shipit:

SergioRAgostinho commented:
Big mistake. Just realized this was tagged for 1.9.0 and not 1.8.1 (facepalm). Oh well...

damienjadeduff commented:
What Could Possibly Go Wrong.

jspricke commented:
This is an ABI change, so we should revert it for 1.8.1. @SergioRAgostinho can you do it?

SergioRAgostinho commented:
That was my concern exactly. There's a big Revert button over there so it I think I can manage. How can we integrate the changes again once we release 1.8.1? Revert from the revert?

jspricke commented:
I guess we need a new pull request. @damienjadeduff can you do that? Thanks!

damienjadeduff commented:
Ok will do.
